OvieMudi/Banka
# #165653259 modify user roles and refactor code

#### What does this PR do?
1. Modifies user role permissions
    - admin can now view transaction history
    - cashier can now activate and deactivate a bank account
    - cashier can now delete a bank account
    - cashier can now update account status
2. Removes password from the returned JSON of the signin page
3. Add test cases to endpoint for better coverage
4. Removes redundant conditional blocks and try-catches

#### Description of Task to be completed?
Ensure the endpoint below does not return a password
```
POST  api/v1/auth/signup
```

#### How should this be manually tested?
- clone the repo ` git clone git@github.com:OvieMudi/Banka.git `

- on your terminal run ` cd Banka `

- run ` git checkout ch-165653259-modify-user-roles-and-refactor `

- run ` npm install `

- run ` npm run dev:start `

- make a request to POST ` localhost:3000/api/v1/auth/signin `

NB see screenshot below on the data to expect

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#165653259](https://www.pivotaltracker.com/story/show/165653259)
#### Screenshots (if appropriate)
![signin_remove password](https://user-images.githubusercontent.com/42499892/56847452-0606ae00-68d3-11e9-902e-f5cf0a55724b.PNG)
